### PR TITLE
Area renderAsIs, refresh props

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Update Area props, pass missing data from hooks, remove dependency on the generator.

## Related Issue

https://github.com/recharts/recharts/issues/4583

### CODE REVIEW

**Comment by PavelVanecek** [comment]
I cannot run the visual diff because I am getting "✖ Failed to authenticate" error.

File: src/cartesian/Area.tsx [hunk 1]
```diff
@@ -533,8 +616,47 @@ function AreaImpl(props: Props) {
...
+  const { height, width, left, top } = useOffset();
+  const updateId = useUpdateId();
+
+  if (layout !== 'horizontal' && layout !== 'vertical') {
+    // Can't render Area in an unsupported layout
+    return null;
+  }
+
+  if (chartName !== 'AreaChart' && chartName !== 'ComposedChart') {
+    // There is nothing stopping us from rendering Area in other charts, except for historical reasons. Do we want to allow that?
```
Comment (ckifer) [inline src/cartesian/Area.tsx:628]: I don't think for this release, maybe eventually everything becomes ComposedChart

**Review by ckifer — APPROVED** [review]
Will make sure to look at the diff

**Comment by PavelVanecek** [comment]
Diff looks good to me: https://www.chromatic.com/build?appId=63da8268a0da9970db6992aa&number=1301

**Comment by ckifer** [comment]
Sweet, I accepted the one with the legend diff

**Pull request merged**